scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -137,16 +141,49 @@ class RunCoverage(
         println("\nComputed Coverage Ratio is: $computedCoverageRatio")
         println("\nGenerated report at: $reportOutputPath\n")
       }
-    } ?: println("No coverage reports generated.")
-  }
 
-  private fun runCoverageForTarget(testTarget: String): CoverageReport {
-    return runBlocking {
-      CoverageRunner(rootDirectory, scriptBgDispatcher, commandExecutor)
-        .runWithCoverageAsync(testTarget.removeSuffix(".kt"))
-        .await()
+      println("COVERAGE ANALYSIS COMPLETED.")
     }
   }
+
+  private fun runCoverageForTarget(testTarget: String): Deferred<CoverageReport> {
+    return CoverageRunner(rootDirectory, scriptBgDispatcher, commandExecutor)
+      .runWithCoverageAsync(testTarget.removeSuffix(".kt"))
+  }
+}
+
+private fun calculateAggregateCoverageReport(
+  coverageReports: List<CoverageReport>
+): CoverageReport {
+  fun aggregateCoverage(coverages: List<Coverage>): Coverage {
+    return if (coverages.contains(Coverage.FULL)) Coverage.FULL
+    else Coverage.NONE
+  }
+
+  val allCoveredLines = coverageReports.flatMap { it.coveredLineList }
+
+  val groupedCoveredLines = allCoveredLines.groupBy { it.lineNumber }
+
+  val aggregatedCoveredLines = groupedCoveredLines.map { (lineNumber, coveredLines) ->
+    CoveredLine.newBuilder()
+      .setLineNumber(lineNumber)
+      .setCoverage(aggregateCoverage(coveredLines.map { it.coverage }))
+      .build()
+  }
+
+  val totalLinesFound = aggregatedCoveredLines.size
+  val totalLinesHit = aggregatedCoveredLines.count { it.coverage == Coverage.FULL }
+
+  val aggregatedTargetList = coverageReports.joinToString(separator = ", ") { it.bazelTestTarget }
+
+  return CoverageReport.newBuilder()
+    .setBazelTestTarget(aggregatedTargetList)
+    .setFilePath(coverageReports.first().filePath)
+    .setFileSha1Hash(coverageReports.first().fileSha1Hash)
+    .addAllCoveredLine(aggregatedCoveredLines)
+    .setLinesFound(totalLinesFound)
+    .setLinesHit(totalLinesHit)
+    .build()
 }
 
 private fun findTestFile(repoRoot: String, filePath: String): List<String> {
```

Comment:
``findTestFiles`` perhaps, since it returns a list?

Reply:
Yeah! It could be a single test file (for non app modules), but it would still be a list, so as suggested using the plural version would be most appropriate Thanks, updated it to `findTestFiles`.